src/main/java/com/mraof/minestuck/client/gui/SummonerScreen.java

```diff
@@ -0,0 +1,114 @@
+		this.te = te;
+		this.summonRange = te.getSummonRange() > 0 ? te.getSummonRange() : 8; //if its defaulted on creation to 0, set it to the intended default of 8
+		this.isUntriggerable = te.getBlockState().getValue(SummonerBlock.UNTRIGGERABLE);
+	}
+	
+	@Override
+	public void init()
+	{
+		int yOffset = (this.height / 2) - (guiHeight / 2);
+		
+		addButton(incrementButton = new ExtendedButton(this.width / 2 + 20, (height - guiHeight) / 2 + 12, 20, 20, new StringTextComponent("+"), button -> changeRange(1)));
+		addButton(decrementButton = new ExtendedButton(this.width / 2 - 40, (height - guiHeight) / 2 + 12, 20, 20, new StringTextComponent("-"), button -> changeRange(-1)));
+		
+		this.entityTypeTextField = new TextFieldWidget(this.font, this.width / 2 - 60, yOffset + 40, 120, 18, new StringTextComponent("Current Entity Type"));	//TODO Use translation instead, and maybe look at other text fields for what the text should be
+		this.entityTypeTextField.setValue(EntityType.getKey(te.getSummonedEntity()).toString());
+		addButton(entityTypeTextField);
+		
+		if(isUntriggerable)
+			addButton(unTriggerableButton = new ExtendedButton(this.width / 2 - 65, yOffset + 70, 85, 20, new StringTextComponent("UNTRIGGERABLE"), button -> cycleUntriggerable()));
+		else
+			addButton(unTriggerableButton = new ExtendedButton(this.width / 2 - 65, yOffset + 70, 85, 20, new StringTextComponent("TRIGGERABLE"), button -> cycleUntriggerable()));
+		addButton(new ExtendedButton(this.width / 2 + 25, yOffset + 70, 40, 20, new StringTextComponent("DONE"), button -> finish()));
+	}
+	
+	/**
+	 * Attempts to increase or decrease the range at which the entity can be summoned if that value is between the min and max
+	 */
+	private void changeRange(int change)
+	{
+		summonRange = MathHelper.clamp(summonRange + change, 1, 64);
+	}
+	
+	/**
+	 * Cycles between the block being triggerable and untriggerable
+	 */
+	private void cycleUntriggerable()
+	{
+		isUntriggerable = !isUntriggerable;
+		if(isUntriggerable)
+			unTriggerableButton.setMessage(new StringTextComponent("UNTRIGGERABLE"));
+		else
+			unTriggerableButton.setMessage(new StringTextComponent("TRIGGERABLE"));
+	}
+	
+	/**
+	 * Returns the current entity type, with imps as the generic entity type
+	 */
+	private EntityType<?> getEntityType(String stringInput)
+	{
+		Optional<EntityType<?>> attemptedEntityType = EntityType.byString(stringInput);
+		return attemptedEntityType.orElse(MSEntityTypes.IMP);
+	}
+	
+	@Override
+	public void render(MatrixStack matrixStack, int mouseX, int mouseY, float partialTicks)
+	{
+		this.renderBackground(matrixStack);
+		RenderSystem.color4f(1F, 1F, 1F, 1F);
+		this.minecraft.getTextureManager().bind(guiBackground);
+		int yOffset = (this.height / 2) - (guiHeight / 2);
+		
+		this.blit(matrixStack, (this.width / 2) - (guiWidth / 2), yOffset, 0, 0, guiWidth, guiHeight);
+		font.draw(matrixStack, Integer.toString(summonRange), (width / 2) - 5, (height - guiHeight) / 2 + 16, 0x404040);
```

Comment:
You have already defined yOffset. Use it instead of `(height - guiHeight) / 2` (and consider if you want to write `yOffset = (height - guiHeight) / 2` instead of how yOffset is written right now)